lib/jekyll/jekyll-sitemap.rb

```diff
@@ -46,6 +46,7 @@ def sitemap
       site_map.content = File.read(source_path).gsub(MINIFY_REGEX, "")
       site_map.data["layout"] = nil
       site_map.data["static_files"] = static_files.map(&:to_liquid)
+      site_map.data["xsl"] = File.exist? @site.in_source_dir("sitemap.xsl")
```

Comment:
I'd abstract `sitemap_exists` to be `file_exists?` and share the logic between the two methods.